partiql-types/src/main/java/org/partiql/value/PartiQLCursor.java

```diff
@@ -0,0 +1,250 @@
+
+    /**
+     * This is applicable to the following types:
+     * {@link PartiQLValueType#BOOL}
+     * @return a value representing the applicable PartiQL value
+     * @throws UnsupportedOperationException when this method is not applicable to the type returned by {@link PartiQLCursor#getType()}
+     * @throws NullPointerException if this method is invoked when {@link PartiQLCursor#isNull()} returns true
+     */
+    public boolean getBoolean() throws UnsupportedOperationException, NullPointerException;
+
+    /**
+     * This is applicable to the following types:
+     * {@link PartiQLValueType#BINARY}
+     * @return a value representing the applicable PartiQL value
+     * @throws UnsupportedOperationException when this method is not applicable to the type returned by {@link PartiQLCursor#getType()}
+     * @throws NullPointerException if this method is invoked when {@link PartiQLCursor#isNull()} returns true
+     * @apiNote <b>! ! ! EXPERIMENTAL ! ! !</b> This is an experimental API under development by the PartiQL maintainers.
+     * Please abstain from using this API until given notice otherwise. This may break between iterations without prior notice.
+     * @deprecated BINARY doesn't exist in SQL or Ion. This is subject to deletion.
+     */
+    @Deprecated
+    @NotNull
+    public byte[] getBytes() throws UnsupportedOperationException, NullPointerException;
+
+    /**
+     * This is applicable to the following types:
+     * {@link PartiQLValueType#BLOB}
+     * @return a value representing the applicable PartiQL value
+     * @throws UnsupportedOperationException when this method is not applicable to the type returned by {@link PartiQLCursor#getType()}
+     * @throws NullPointerException if this method is invoked when {@link PartiQLCursor#isNull()} returns true
+     * @apiNote <b>! ! ! EXPERIMENTAL ! ! !</b> This is an experimental API under development by the PartiQL maintainers.
+     * Please abstain from using this API until given notice otherwise. This may break between iterations without prior notice.
+     */
+    @NotNull
+    public Blob getBlob() throws UnsupportedOperationException, NullPointerException;
+
+    /**
+     * This is applicable to the following types:
+     * {@link PartiQLValueType#CLOB}
+     * @return a value representing the applicable PartiQL value
+     * @throws UnsupportedOperationException when this method is not applicable to the type returned by {@link PartiQLCursor#getType()}
+     * @throws NullPointerException if this method is invoked when {@link PartiQLCursor#isNull()} returns true
+     * @apiNote <b>! ! ! EXPERIMENTAL ! ! !</b> This is an experimental API under development by the PartiQL maintainers.
+     * Please abstain from using this API until given notice otherwise. This may break between iterations without prior notice.
+     */
+    @NotNull
+    public Clob getClob() throws UnsupportedOperationException, NullPointerException;
```

Comment:
I'm not sure these are necessary considering you can use `getBytes()`. I would remove them for the time being so this package has to java.sql dependency.

Reply:
ACK. Just updated to remove for now.